mock-etcd/src/mock_etcd.rs

```diff
@@ -1388,45 +1405,51 @@ mod test {
     fn e2e_lock_lease_test(client: &Arc<Client>) {
         log::debug!("start e2e_lock_lease_test");
         smol::future::block_on(async {
-            let lock_key012 = vec![0_u8, 1_u8, 2_u8];
-            let res = client
-                .lease()
-                .grant(EtcdLeaseGrantRequest::new(std::time::Duration::from_secs(
-                    5,
-                )))
-                .await
-                .unwrap_or_else(|err| panic!("failed to get key 0, the error is {}", err));
-            let lease_id = res.id();
-
-            // lock unlock twice
-            {
-                log::debug!("test basic lock");
-                let mut res = client
-                    .lock()
-                    .lock(EtcdLockRequest::new(lock_key012.clone(), lease_id))
-                    .await
-                    .unwrap_or_else(|err| panic!("failed to lock key012, the error is {}", err));
-
-                client
-                    .lock()
-                    .unlock(EtcdUnlockRequest::new(res.take_key()))
+            async fn test_lock_basic(client: &Client) {
```

Comment:
Why do we defind functions in a closure and test them in a async task?

Reply:
Just divide the test to three part, make the code more clear.